pkg/audit/manager.go

```diff
@@ -507,45 +509,26 @@ func (am *Manager) addAuditResponsesToUpdateLists(
 	return nil
 }
 
-func (am *Manager) writeAuditResults(ctx context.Context, resourceList []schema.GroupVersionKind, updateLists map[string][]auditResult, timestamp string, totalViolations map[string]int64) error {
-	// get constraints for each Kind
-	for _, constraintGvk := range resourceList {
-		am.log.Info("constraint", "resource kind", constraintGvk.Kind)
-		instanceList := &unstructured.UnstructuredList{}
-		instanceList.SetGroupVersionKind(constraintGvk)
-		err := am.client.List(ctx, instanceList)
-		if err != nil {
-			return err
-		}
-		am.log.Info("constraint", "count of constraints", len(instanceList.Items))
+func (am *Manager) writeAuditResults(ctx context.Context, resourceList []schema.GroupVersionKind, updateLists map[string][]auditResult, timestamp string, totalViolations map[string]int64) {
+	am.ucloop = &updateConstraintLoop{
+		client:  am.client,
+		stop:    make(chan struct{}),
+		stopped: make(chan struct{}),
+		ul:      updateLists,
+		ts:      timestamp,
+		tv:      totalViolations,
+		log:     am.log,
+	}
 
-		updateConstraints := make(map[string]unstructured.Unstructured, len(instanceList.Items))
-		// get each constraint
-		for _, item := range instanceList.Items {
-			updateConstraints[item.GetSelfLink()] = item
-		}
-		if len(updateConstraints) > 0 {
-			if am.ucloop != nil {
-				close(am.ucloop.stop)
-				select {
-				case <-am.ucloop.stopped:
-				case <-time.After(time.Duration(*auditInterval) * time.Second):
-				}
-			}
-			am.ucloop = &updateConstraintLoop{
-				uc:      updateConstraints,
-				client:  am.client,
-				stop:    make(chan struct{}),
-				stopped: make(chan struct{}),
-				ul:      updateLists,
-				ts:      timestamp,
-				tv:      totalViolations,
-			}
-			am.log.Info("starting update constraints loop", "updateConstraints", updateConstraints)
-			go am.ucloop.update()
+	if am.ucloop.uc != nil {
+		close(am.ucloop.stop)
```

Comment:
nit: I wonder if we want to add a log line for when we need to explicitly kill the reporting loop. That would allow us to more easily diagnose incomplete reports caused by too-frequent audits.

Reply:
added